language_service/src/signature_help/tests.rs

```diff
@@ -257,7 +256,48 @@ fn insert_second_argument() {
             }
         }
     "#},
-        &expect![[r#""#]],
+        &expect![[r#"
+            SignatureHelp {
+                signatures: [
+                    SignatureInformation {
+                        label: "operation Foo(x : Int, y : Double, z : String) : Unit",
+                        documentation: None,
+                        parameters: [
+                            ParameterInformation {
+                                label: Span {
+                                    start: 13,
+                                    end: 46,
+                                },
+                                documentation: None,
+                            },
+                            ParameterInformation {
+                                label: Span {
+                                    start: 14,
+                                    end: 21,
+                                },
+                                documentation: None,
+                            },
+                            ParameterInformation {
+                                label: Span {
+                                    start: 23,
+                                    end: 33,
+                                },
+                                documentation: None,
+                            },
+                            ParameterInformation {
+                                label: Span {
+                                    start: 35,
+                                    end: 45,
+                                },
+                                documentation: None,
+                            },
+                        ],
+                    },
+                ],
+                active_signature: 0,
+                active_parameter: 3,
```

Comment:
This seems like a bug... shouldn't this be `2` and not `3`? Playing with it in the UI, it does seem to give help for the next item until I start actually typing and then it jumps back to the current one. This may be a bug in how I did `Err` handling in the signature help...

Reply:
@minestarks comment on the spans ended up being the key here, and now this is working as expected.